python_modules/libraries/dagster-datadog/dagster_datadog/resources.py

```diff
@@ -0,0 +1,59 @@
+from datadog import initialize, statsd
+
+from dagster import Dict, Field, ResourceDefinition, String
+
+
+class DataDogResource:
+    def __init__(self, api_key, app_key):
+        self.api_key = api_key
+        self.app_key = app_key
+        initialize(api_key=api_key, app_key=app_key)
+
+    def event(
+        self,
+        title,
+        text,
+        alert_type=None,
+        aggregation_key=None,
+        source_type_name=None,
+        date_happened=None,
+        priority=None,
+        tags=None,
+        hostname=None,
+    ):
+        statsd.event(
+            title=title,
+            text=text,
+            alert_type=alert_type,
+            aggregation_key=aggregation_key,
+            source_type_name=source_type_name,
+            date_happened=date_happened,
+            priority=priority,
+            tags=tags,
+            hostname=hostname,
+        )
+
+    def increment(self, key, value=1, tags=None, sample_rate=1):
+        statsd.increment(key, value, tags, sample_rate)
+
+    def decrement(self, key, value=1, tags=None, sample_rate=1):
+        statsd.decrement(key, value, tags, sample_rate)
+
+    def gauge(self, key, value, tags=None, sample_rate=1):
+        statsd.gauge(key, value, tags, sample_rate)
+
+    def histogram(self, key, value, tags=None, sample_rate=1):
+        statsd.histogram(key, value, tags, sample_rate)
+
+    def set(self, metric, value, tags=None, sample_rate=1):
+        statsd.set(metric, value, tags, sample_rate)
```

Comment:
is this the entire surface area of statsd? I'm generally always a fan of generic passthrough for these types of things

Reply:
good call, there were a couple I omitted, fixed.

Reply:
I guess what i'm saying is that it seems like we should have an explicitly supported way of calling through to statsd so that we if someone doesn't keep this wrapper in sync with the underlying API it isn't some sort of crisis